Neo4j.Driver/Neo4j.Driver/Internal/Extensions/ErrorExtensions.cs

```diff
@@ -31,56 +36,7 @@ public static Neo4jException ParseServerException(string code, string message)
             switch (classification)
             {
                 case "clienterror":
-                    if (AuthenticationException.IsAuthenticationError(code))
-                    {
-                        error = new AuthenticationException(message);
-                    }
-					else if(AuthorizationException.IsAuthorizationError(code))
-					{
-						error = new AuthorizationException(message);
-					}
-                    else if (ProtocolException.IsProtocolError(code))
-                    {
-                        error = new ProtocolException(code, message);
-                    }
-                    else if (FatalDiscoveryException.IsFatalDiscoveryError(code))
-                    {
-                        error = new FatalDiscoveryException(message);
-                    }
-					else if(TokenExpiredException.IsTokenExpiredError(code))
-					{
-						error = new TokenExpiredException(message);
-					}
-					else if(InvalidBookmarkException.IsInvalidBookmarkException(code))
-					{
-						error = new InvalidBookmarkException(message);
-					}
-                    else if (InvalidBookmarkMixtureException.IsInvalidBookmarkMixtureException(code))
-                    {
-                        error = new InvalidBookmarkMixtureException(message);
-                    }
-                    else if (ArgumentErrorException.IsArgumentErrorException(code))
-                    {
-                        error = new ArgumentErrorException(message);
-                    }
-                    else if (TypeException.IsTypeException(code))
-                    {
-                        error = new TypeException(message);
-                    }
-                    else if (ForbiddenException.IsForbiddenException(code))
-                    {
-                        error = new ForbiddenException(message);
-                    }
-                    // this one needs to come after it has checked all other possibilities
-                    else if (UnknownSecurityException.IsUnknownSecurityException(code))
-                    {
-                        return new UnknownSecurityException(message, code);
-                    }
-                    else
-                    {
-                        error = new ClientException(code, message);
-                    }
-
+                    error = _clientErrorExceptionFactory.GetException(code, message);
                     break;
                 case "transienterror":
```

Comment:
if you gave transientexception a "Neo.TransientError.*" attribute, could we remove this switch?
if no thing matches client/transient it is a database exception

Reply:
we could use regex, and put negative patterns into support database exception too as that would look like `^\w+\.(?!(clienterror|transienterror)).*`